File: solutions_5738606668808192_0/C++/shyam23vinay/GCJ_coinjam.cpp
#include<bits/stdc++.h>
using namespace std;

void next(int c[],int n)
{
    for(int i=n-2;i>=1;i--)
    {
        if(c[i]==1)
            c[i]=0;
        else
        {
            c[i]=1;
            break;
        }
    }
}

unsigned long long int res(unsigned long long int n)
{
    for(unsigned long long int i=2;i*i<=n;i++)
    {
        if(n%i==0)
            return i;
    }
    return 1;
}

int main()
{
    ofstream fout ("C-small-attempt0.out");
    ifstream fin ("C-small-attempt0.in");
    int t;
    fin >> t;
    for(int i=1;i<=t;i++)
    {
        int n,j;
        fin >> n >> j;
        fout << "Case #"<<i<<":"<<endl;
        int a[n+1];
        a[0]=1;
        a[n-1]=1;
        for(int k=1;k<=n-2;k++)
            a[k]=0;
        for(int k=1;k<=j;next(a,n))
        {
            int flag=0;
            vector<unsigned long long int>v;
            unsigned long long int ten=0;
            for(unsigned long long int p=0;p<n;p++)
            {
                ten=ten*10+a[p];
            }
            v.push_back(ten);
            for(unsigned long long int x=2;x<=10;x++)
            {
                unsigned long long int temp=0;
                for(unsigned long long int p=0;p<n;p++)
                {
                    temp=temp*x+a[p];
                }
                unsigned long long int temp_res=res(temp);
                if(temp_res==1)
                {
                    flag=1;
                    break;
                }
                else
                {
                    v.push_back(temp_res);
                }
            }
            if(flag==0)
            {
                for(int p=0;p<v.size();p++)
                    fout << v[p] << " ";
                fout << endl;
                k++;
            }
        }
    }
    return 0;
}

File: solutions_5738606668808192_1/Python/shyam23vinay/gcj_coinjam.py
def res(n):
    i=2
    while(i<=10000):
        if(n%i==0):
            return i
        i=i+1
    return 1

def nxt(a):
    l=len(a)
    s=['1']*l
    k=l-2
    index=1
    while(k>=1):
        if(a[k]=='1'):
            s[k]='0'
        else:
            s[k]='1'
            index=k
            break
        k=k-1
    index=index-1
    while(index>=1):
        s[index]=a[index]
        index=index-1
    st=''
    for e in s:
        st=st+e
    return st

fin=file("C-large-attempt0.in","r")
fout=open("C-large-attempt0.out","w")
fin=fin.read()
fin=fin.split('\n')
t=int(fin[0])
for tc in range(1,t+1):
    n,j=fin[tc].split(" ")
    n=int(n)
    j=int(j)
    ans=''
    ans=ans+"Case #"+str(tc)+":\n"
    s='1'+'0'*(n-2)+'1'
    k=1
    while(k<=j):
        a=[]
        flag=0
        ten=0
        for p in range(0,n):
            ten=ten*10+int(s[p])
        a.append(ten)
        for x in range(2,11):
            temp=0
            for p in range(0,n):
                temp=temp*x+int(s[p])
            temp_res=res(temp)
            if(temp_res==1):
                flag=1
                break
            else:
                a.append(temp_res)
        if(flag==0):
            for p in range(0,len(a)):
                ans=ans+ str(a[p])+" "
            ans=ans+"\n"
            k=k+1
        s=nxt(s)
fout.write(ans)
fout.close()

